### ISSUE TITLE
`JsonParser.Feature.EMPTY_STRING_AS_NULL` does not work when text is parsed as `String[]`

### ISSUE THREAD

**Issue by wkwkhautbois** [FasterXML/jackson-dataformats-text#222]
`CsvParser.Feature.EMPTY_STRING_AS_NULL` not works when csv is parsed `String[]`.
On the other hand, the feature is valid when csv is parsed as `Object[]`.

```java
String csv = "Grace,,Hopper";

CsvMapper csvMapper = CsvMapper.builder()
        .enable(CsvParser.Feature.EMPTY_STRING_AS_NULL)
        .enable(CsvParser.Feature.WRAP_AS_ARRAY)
        .build();

MappingIterator<Object[]> it1 =  csvMapper.readerFor(Object[].class).readValues(csv);
it1.hasNext();
Object[] array1= it1.next();

MappingIterator<String[]> it2 =  csvMapper.readerFor(String[].class).readValues(csv);
it2.hasNext();
String[] array2= it2.next();

System.out.println(Arrays.toString(array1));    // [Grace, null, Hopper]
System.out.println(Arrays.toString(array2));    // [Grace, , Hopper]      array2[1] is not null, but empty string ""
```

This is strange behavior and both case should parsed as `null`, I think.
Is this a bug?

jackson v.2.11.2

**Comment by cowtowncoder** [comment]
@wkwkhautbois it does sound like a bug. Thank you for reporting this.

**Comment by cowtowncoder** [comment]
Interesting. Yes, I can reproduce this; due to missing handling in `CsvParser.nextTextValue()` (and possibly databind's optimizations for `String[]` type, but we'll see about that). I think that there's some risk of regression so will target fix for 2.12.

**Comment by wkwkhautbois** [comment]
Thank you, I'm looking forward to next version release!
